### DISCUSSION TITLE
How to use a camera with Create 3 for ROS 2

### DISCUSSION THREAD

**Discussion by AsawareeBhide** [iRobotEducation/create3_docs#293] [Questions]
Hi! In the context of using a camera with ROS 2 -

1. Can a camera be connected directly to Create 3 via the USB-C connector or is a compute board needed in between?
2. If connected directly to Create 3, can the camera communicate with a compute board over WiFi like Create 3 does (using its ROS 2 wrapper)?

**Comment by alsora** [comment]
Long story short: you will need a compute board.

If you connect the camera to the create 3, the camera may power up but you will not have a way to run the camera driver software (you can't run custom software on the create 3 robot; moreover even if you could, this would likely be too computationally intensive for the robot processor). 

There are similar questions in this forum that you can look at:
 - https://github.com/iRobotEducation/create3_docs/discussions/127
 - https://github.com/iRobotEducation/create3_docs/discussions/90
